internal/release/promote/promotion.go

```diff
@@ -131,20 +131,23 @@ func (p *Promotion) Promote(opts Opts) (string, error) {
 	if err != nil {
 		return "", fmt.Errorf("selecting releases to promote: %w", err)
 	}
-	if !list.HasAnyPromotableReleases() {
-		p.promptProvider.PrintNoPromotableReleasesFound(opts.ReleasesFiltered, opts.SourceEnv, opts.TargetEnv)
+
+	invalidList := selectedList.GetNonPromotableReleases(opts.SourceEnv, opts.TargetEnv)
+	if len(invalidList) != 0 {
+		invalid := strings.Join(invalidList, ", ")
+		p.promptProvider.PrintSelectedNonPromotableReleases(invalid, opts.TargetEnv.Name)
 		return "", nil
```

Comment:
We should not be returning a nil error. We want the program to fail if invalid releases are selected